src/lib/RpcServer.es.js

```diff
@@ -119,7 +133,7 @@ class RpcState {
         this._postMessage = 'source' in message && !('returnURL' in message);
         this._returnURL = 'returnURL' in message ? message.returnURL : null;
         this._data = message.data;
-        this._source = 'source' in message ? message.source : null;
+        this._source = 'source' in message ? /** @type {Window} */ (message.source) : null;
```

Comment:
Do you know at this point that source is a window, or is it just your expectation? :)

Reply:
Since the Keyguard should never receive a valid postMessage from a Worker or any context other than a parent `window`, I know it.